src/Analysis_Clustering.cpp

```diff
@@ -228,6 +229,10 @@ Analysis::RetType Analysis_Clustering::Setup(ArgList& analyzeArgs, DataSetList*
   // Output option for cluster info
   suppressInfo_ = analyzeArgs.hasKey("noinfo");
 
+  // Dataset to store representative cluster frames
+  clustersRefFrame_ = (DataSet_integer*) datasetlist->AddSet(DataSet::INTEGER, "refframe", "refframe");
+  if (clustersRefFrame_==0) return Analysis::ERR;
+
   // Dataset to store cluster number v time
```

Comment:
Can you make another PR for this? I will close this one rather pushing new commit. It is easier for both of us. :d

Hai

> On Nov 23, 2015, at 11:41 AM, Daniel R. Roe <notifications@github.com> wrote:
> 
> In src/Analysis_Clustering.cpp:
> 
> > @@ -228,6 +229,10 @@ Analysis::RetType Analysis_Clustering::Setup(ArgList& analyzeArgs, DataSetList*
> >    // Output option for cluster info
> >    suppressInfo_ = analyzeArgs.hasKey("noinfo");
> >  
> > +  // Dataset to store representative cluster frames
> > +  clustersRefFrame_ = (DataSet_integer*) datasetlist->AddSet(DataSet::INTEGER, "refframe", "refframe");
> > +  if (clustersRefFrame_==0) return Analysis::ERR;
> > +
> >    // Dataset to store cluster number v time
> This should come after the cluster num vs time data set is allocated and then use that DataSet name. All DataSets generated by an Action or an Analysis should have the same name. In this case you want to use the name of the cnumvtime_ DataSet and add the Aspect 'refframe', e.g.
> 
> clustersRefFrame_ = datasetlist->AddSet(DataSet::Integer, MetaData(cnumvtime_->Meta().Name(), "refframe"));
> Also, you don't need to make clustersRefFrame_ a pointer to DataSet_integer. Just make it a DataSet*. Then you avoid having the DataSet_integer.h include in Analysis_Clustering.h and make things a bit more general. You can always cast to DataSet_integer inside Analysis_Clustering.cpp if need be.
> 
> —
> Reply to this email directly or view it on GitHub.
>